File: Aigilas/Aigilas/States/GameplayState.cs
using SPX.States;
using SPX.Entities;
using Aigilas.Creatures;
using Aigilas.Dungeons;
using SPX.Core;
using System;
using SPX.IO;

namespace Aigilas.States
{
    public class GameplayState : State
    {
        public GameplayState()
        {
            Console.WriteLine("Generating the dungeon...");
            EntityManager.Reset();
            DungeonFactory.Start();
            Client.Get().DungeonHasLoaded();
        }
        public void Update()
        {
            EntityManager.Update();
        }
        public void LoadContent()
        {
            EntityManager.LoadContent();
        }
        public void Draw()
        {
            EntityManager.Draw();
        }
    }
}

File: proj/src/aigilas/states/GameplayState.java
package aigilas.states;

import aigilas.dungeons.DungeonFactory;
import spx.entities.EntityManager;
import spx.entities.IActor;
import spx.net.Client;
import spx.particles.ParticleEngine;
import spx.states.State;
import spx.states.StateManager;
import spx.text.TextManager;

import java.util.ArrayList;
import java.util.List;

public class GameplayState implements State {
    public GameplayState() {
        System.out.println("Generating the dungeon...");
        EntityManager.reset();
        DungeonFactory.start();
        Client.get().dungeonHasLoaded();
    }

    List<IActor> players = new ArrayList<>();
    @Override
    public void update() {
        players = EntityManager.getPlayers();
        boolean allDead = true;
        for(IActor player:players){
            if(player.isActive()){
                allDead = false;
            }
        }
        if(allDead){
            StateManager.loadState(new GameOverState());
        }
        EntityManager.update();
    }

    @Override
    public void loadContent() {
        EntityManager.loadContent();
    }

    @Override
    public void unload() {
        EntityManager.clear();
        ParticleEngine.reset();
        TextManager.clear();
    }

    @Override
    public void draw() {
        EntityManager.draw();
    }
}
